tests/test_engine.py

```diff
@@ -52,6 +52,18 @@ def test_get_lookup_url(self):
         assert final_uri_with_params == '1/2/?extra_param=3'
         SampleResourceModel._lookup_urls = []
 
```

Comment:
I tested in a shell and using literals like this causes the items to end up in the dictionary in different orders. Should I test some other values as well? How does this look?